### DISCUSSION TITLE
Do not pass events that are not accepted

### DISCUSSION THREAD

**Discussion by fanouris14** [derekantrican/GAS-ICS-Sync#368] [General]
hello everyone,

is there any way to sync events that i have not accepted yet and show them 'correctly'?

images bellow:
![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/a6507549-5d6b-4cba-baeb-e9a4e32518e3)
![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/aade8745-eb07-4a85-9def-2c3029a57b47)
![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/70c7fc58-6231-41d8-8b9c-69dfeaaf1a24)

in case i have not accepted the invitation, i would like to show the event 'hollow'

thank you!

**Comment by jonas0b1011001** [comment]
What's your setup? Google Calendar A is invited to an event, and the script syncs Google Calendar A to Google Calendar B?

**Reply by fanouris14** [reply]
hello there! thanks for the reply!

i have an outlook Calendar and i've send an invitation though mail to my google callendar. i've used the script to my google calendar to 'subscribe' to that external outlook calendar

**Reply by jonas0b1011001** [reply]
I would assume the event gets automatically created in gcal after you send the invite, is that not the case?

**Reply by fanouris14** [reply]
i will try to explain the flow a little bit more in detail.

everything starts from the outlook calendar. i get new invites, in some cases i accept them, and in others i don't answer 'yes' neither 'no'. so in the end i end up with something like this into the outlook calendar:

![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/2ddb7a8a-0d4a-4a6d-9102-bfb7e0fab49f)
(event on the left: hollow because i did not accept/decline)
(event on the right: accepted)

so in the end, using the script i get all events and they all show up in my google callendar, as 'accepted'

**Reply by jonas0b1011001** [reply]
The acceptance feature will only work for events the gmail account was invited to, we have no control over it.

If you sync a fake event with your gmail account as participant, you will get the 'host' event in your targetCalendar (blue) + a 'hollow' event (as you described above) in your main calendar (orange).

```
BEGIN:VEVENT
DTSTART:20230930T080000Z
DTEND:20230930T083000Z
DTSTAMP:20230928T182507Z
ATTENDEE;CUTYPE=INDIVIDUAL;ROLE=OPT-PARTICIPANT;PARTSTAT=NEEDS-ACTION;CN=jonas0b1011001@gmail.com;X-NUM-GUESTS=0:mailto:jonas0b1011001@gmail.com
CREATED:20230928T182301Z
LAST-MODIFIED:20230928T182434Z
SEQUENCE:1
STATUS:CONFIRMED
SUMMARY:test
TRANSP:TRANSPARENT
END:VEVENT
```
<img width="196" alt="Bildschirmfoto 2023-09-29 um 19 44 01" src="https://github.com/derekantrican/GAS-ICS-Sync/assets/43352574/2892ace9-a010-40cc-9b40-12d23d247edb">

**Reply by fanouris14** [reply]
thank you for the detailed answer,

so is there any way to not sync at all events that i have not yet accepted in my main account? and sync them only if i accept them [from my outlook] ?

**Reply by jonas0b1011001** [reply]
Try adding this to [helpers.gs line 184](https://github.com/derekantrican/GAS-ICS-Sync/blob/9990d1b037bbc3e2d72f130f8a815c0f71bd3bcd/Helpers.gs#L184):
  ```
result = result.filter(function(event){
    if (!event.hasProperty('attendee'))
      return true;
    
    let result = true; 
    let attendees = event.getAllProperties('attendee');
    for (attendee of attendees){
      let mail = attendee.getParameter('cn');
      let response = attendee.getParameter('partstat');
      if (mail == "<<add your outlook mail>>" && response != "ACCEPTED"){
        result = false;
        break;
      }
    }
    return result;
  });
```
Don't forget to edit the mail address in the middle of the new code.

**Reply by fanouris14** [reply]
oh thank you for the snippet, i ll try it at once. by mail adress you mean the ics url? or the real mail? 'test@test.com"

**Reply by jonas0b1011001** [reply]
The real mail address that get's invited to the events.

**Reply by fanouris14** [reply]
well i tried the script, i added the code at the correct line. saved the file, unistall the previous running script, removed the ics callendar from the google account and finally run the script again from start. i had some past events that i had not accepted them, but the script synced them.

![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/a12870a0-841e-4e93-9d0c-ceed367fbc39)
![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/3ed56c3c-abd2-4225-b0ee-390281f8731e)

**Reply by jonas0b1011001** [reply]
Can you share one of the events?

- open the *.ics file in a texteditor
- search for the event title
- post everything between BEGIN:VEVENT and END:VEVENT (you can remove any sensitive data of course)

**Reply by fanouris14** [reply]
ofc!

![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/9dc0eb41-b483-47f9-9f4b-6e532109d4b6)

**Reply by jonas0b1011001** [reply]
Can you confirm that events you accepted have set `X-MICROSOFT-CDO-BUSYSTATUS:BUSY` instead of `TENTATIVE`?

**Reply by fanouris14** [reply]
this seems to be true, this is an event i accepted:

![image](https://github.com/derekantrican/GAS-ICS-Sync/assets/101338931/23d47d0e-d069-4199-b7bc-cbdb80054b09)

**Reply by jonas0b1011001** [reply]
ok, use this instead of the code above:
```
result = result.filter(function(event){
    try{
      return (event.getFirstPropertyValue('x-microsoft-cdo-busystatus').toString().toLowerCase() == "busy");
    }catch(e){
      return true;
    }
  });
```

**Reply by fanouris14** [reply]
oh my god! now it works as i wanted! thank you so much!
